### PR TITLE
Feat/dat 2384 save es builds output final mappings in gdc models

### PR DESCRIPTION
**Pull request by jesuspguofc**

### CODE REVIEW

File: gdcmodels/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import yaml
 from os import listdir
 import pkg_resources
-from os.path import isfile, join as pj
+from os.path import isfile,dirname, join as pj
```
Comment (anmaxvl) [inline gdcmodels/__init__.py]:
```suggestion
from os.path import isfile, dirname, join as pj
```

File: gdcmodels/__init__.py [hunk 2]
```diff
@@ -11,6 +11,20 @@ def load_yaml(filename):
         return yaml.safe_load(f)
 
 
+def process_ref(name, reference, path):
+    ref = reference['$ref']
```
Comment (anmaxvl) [inline gdcmodels/__init__.py]: this might break for `*_centric` mappings

File: gdcmodels/__init__.py [hunk 3]
```diff
@@ -11,6 +11,20 @@ def load_yaml(filename):
...
+    filename, key = ref.split('#/')
+    filedata = load_yaml(pj(path,filename))
+    return filedata[key] 
+
+
+def load_definitions(filename):
+    import pdb; pdb.set_trace()
```
Comment (anmaxvl) [inline gdcmodels/__init__.py]: 💣

File: gdcmodels/__init__.py [hunk 4]
```diff
@@ -11,6 +11,20 @@ def load_yaml(filename):
         return yaml.safe_load(f)
 
 
+def process_ref(name, reference, path):
+    ref = reference['$ref']
+    filename, key = ref.split('#/')
+    filedata = load_yaml(pj(path,filename))
```
Comment (anmaxvl) [inline gdcmodels/__init__.py]:
```suggestion
    filedata = load_yaml(pj(path, filename))
```

File: gdcmodels/__init__.py [hunk 5]
```diff
@@ -35,7 +49,10 @@ def get_es_models():
                 es_models[es_index][es_type] = {
                         '_mapping': load_yaml(pj(es_model_dir, es_index, f))
                     }
-
+                defs_name = pj(es_model_dir, es_index,es_type+".".join(['_definitions', 'yaml']))
```
Comment (anmaxvl) [inline gdcmodels/__init__.py]:
```suggestion
                defs_name = pj(es_model_dir, es_index, es_type + '_definitions.yaml')
```
don't really see the need of the `join` here

File: gdcmodels/__init__.py [hunk 6]
```diff
@@ -35,7 +49,10 @@ def get_es_models():
...
+                if isfile(defs_name):
+                    definitions = load_definitions(defs_name)
+                    es_models[es_index][es_type]['_mapping'].update(definitions)
```
Comment (anmaxvl) [inline gdcmodels/__init__.py]: if `$ref` is in the file, you'll want to clean it up as well.

File: gdcmodels/__init__.py (hunk 3 above)
Comment (jesuspguofc) [reply gdcmodels/__init__.py]: my bad lol

File: gdcmodels/__init__.py (hunk 5 above)
Comment (jesuspguofc) [reply gdcmodels/__init__.py]: simpler, I like it

File: gdcmodels/__init__.py (hunk 6 above)
Comment (jesuspguofc) [reply gdcmodels/__init__.py]: that's correct thanks

File: gdcmodels/__init__.py (hunk 2 above)
Comment (jesuspguofc) [reply gdcmodels/__init__.py]: humm not sure I see how. right now it is running through them without a blowing up... maybe we can sync up offline

File: gdcmodels/__init__.py (hunk 1 above)
Comment (jesuspguofc) [reply gdcmodels/__init__.py]: yup

File: gdcmodels/__init__.py (hunk 4 above)
Comment (jesuspguofc) [reply gdcmodels/__init__.py]: yup, sorry ahout that

File: gdcmodels/__init__.py (hunk 2 above)
Comment (anmaxvl) [reply gdcmodels/__init__.py]: oh, you are right, there's that `isfile` check. my bad.

File: tests/conftest.py [hunk 7]
```diff
@@ -119,3 +119,75 @@ def mock_mappings(tmp_path):
...
+def mock_isfile_def(filename):
+    if os.path.basename(filename) == 'gdc_from_graph':
+        return False
+    elif os.path.basename(filename) == 'definitions.yaml':
+        return True
+    return True
+
+
+@pytest.fixture
+def add_def_file(monkeypatch):
```
Comment (anmaxvl) [inline tests/conftest.py]: can we mock the filesystem instead of mocking calls? similar to what's being done in `mock_mappings` fixture?

Comment (jesuspguofc) [reply tests/conftest.py]: you got it boss

**Pull request merged**